ManageIQ/manageiq
# Convert ExplorerPresenter to JSON

Purpose or Intent
-----------------
We want to get rid of RJS (and javascript eval). This PR converts all the actions that go through ExplorerPresenter to JSON. This is surely not as good as using some nice JSON/REST API or converting the stuff in question to Angular. But it implements a more restricted interface and removes the need to call `eval` in many situations.

Steps for Testing/QA
--------------------
Every screen that uses explorer (trees on the left) is affected by this change. And some more. I expect bugs. Actually if there would be no bugs found, I would be pretty scared.

Make sure that `miqBuildCalendar` related stuff works well. Note: Seems to me that there's a lot of unused code around that. Needs a cleanup (in a separate PR).
